Explorer/Assets/Scripts/ECS/SceneLifeCycle/Systems/UnloadPortableExperiencesSystem.cs

```diff
@@ -0,0 +1,39 @@
+﻿using Arch.Core;
+using Arch.System;
+using Arch.SystemGroups;
+using ECS.Abstract;
+using ECS.Groups;
+using ECS.LifeCycle.Components;
+using ECS.SceneLifeCycle.SceneDefinition;
+
+namespace ECS.SceneLifeCycle.Systems
+{
+    [UpdateInGroup(typeof(CleanUpGroup))]
+    [UpdateBefore(typeof(UnloadSceneSystem))]
+    public partial class UnloadPortableExperiencesSystem : BaseUnityLoopSystem
+    {
+        internal UnloadPortableExperiencesSystem(World world) : base(world) { }
+
+        protected override void Update(float t)
+        {
+            UnloadPortableExperienceRealmQuery(World);
+        }
+
+        [Query]
+        [All(typeof(DeleteEntityIntention), (typeof(PortableExperienceRealmComponent)))]
+        private void UnloadPortableExperienceRealm(in Entity entity, ref PortableExperienceComponent portableExperienceComponent)
+        {
+            //We start another query from here using the data from the px component to match all other entities that were created by this PX
+            UnloadLoadedPortableExperienceSceneQuery(World, portableExperienceComponent.Ens.ToString());
+            World.Remove<PortableExperienceRealmComponent, PortableExperienceComponent>(entity);
+        }
+
+        [Query]
+        [None(typeof(DeleteEntityIntention))]
+        private void UnloadLoadedPortableExperienceScene([Data] string sceneEntityId, in Entity entity, ref PortableExperienceComponent portableExperienceComponent)
+        {
+            //We only set to destroy those entities that have the same ens than the PX
+            if (portableExperienceComponent.Ens.ToString() == sceneEntityId) World.Add<DeleteEntityIntention>(entity);
```

Comment:
I am a bit confused with this flow: `PortableExperienceComponent` contains own `ENS`, not the parent ENS. Is it correct? Or how does it work?

Reply:
So, I made it so that the scene created by the PX will contain the same ENS used to create the realm, so its easy to match a PX scene to the realm that created it. 
So in the PortableExperiencesController we add the PortableExperienceComponent with the ens used to create the Realm to the same entity that has the PortableExperienceRealmComponent. Later when the scenes are resolved, that component is basically copied into the scene entity (as it also needs a PortableExperienceComponent to filter it in other queries). This way it makes it very easy to iterate and add the DeleteEntityIntention to the correct entities.